projects/hslayers/components/wfs-filter/wfs-filter.component.ts

```diff
@@ -0,0 +1,325 @@
+import {catchError, filter, lastValueFrom, map, switchMap, tap} from 'rxjs';
+import {computed, signal} from '@angular/core';
+import {
+  getWfsAttributes,
+  getWfsUrl,
+  setWfsAttributes,
+} from 'hslayers-ng/common/extensions';
+import {takeUntilDestroyed, toSignal} from '@angular/core/rxjs-interop';
+
+@Component({
+  selector: 'hs-wfs-filter',
+  templateUrl: './wfs-filter.component.html',
+  styles: `
+    .hs-wfs-filter-panel > div {
+      padding: 0 0.75rem;
+    }
+  `,
+  standalone: true,
+  imports: [
+    NgClass,
+    HsFiltersComponent,
+    FormsModule,
+    AsyncPipe,
+    HsPanelHeaderComponent,
+    TranslateCustomPipe,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HsWfsFilterComponent extends HsPanelBaseComponent {
+  name = 'wfsFilter';
+
+  selectedLayer = signal<HsLayerDescriptor | null>(null);
+  rule = computed(() => this.selectedLayer()?.layer.get('wfsFilter') || {});
+
+  hsFiltersService = inject(HsFiltersService);
+  hsEventBusService = inject(HsEventBusService);
+  hsLayerManagerService = inject(HsLayerManagerService);
+  hsUtilsService = inject(HsUtilsService);
+  hsLayoutService = inject(HsLayoutService);
+  httpClient = inject(HttpClient);
+  hsLayerSelectorService = inject(HsLayerSelectorService);
+
+  availableLayers: Signal<HsLayerDescriptor[]>;
+
+  constructor() {
+    super();
+    const mainPanelStream = this.hsEventBusService.mapEventHandlersSet.pipe(
+      switchMap(() => this.hsLayoutService.mainpanel$),
+      filter((which) => which === 'wfsFilter'),
+      takeUntilDestroyed(),
+    );
+
+    this.availableLayers = toSignal(
+      mainPanelStream.pipe(
+        map(() => {
+          return this.hsLayerManagerService.data.layers.filter(
+            (l: HsLayerDescriptor) =>
+              this.hsUtilsService.instOf(l.layer, VectorLayer) &&
+              this.hsUtilsService.instOf(l.layer.getSource(), VectorSource) &&
+              getWfsUrl(l.layer),
+          );
+        }),
+        tap((layers) => {
+          const currentLayerTitle =
+            this.hsLayerSelectorService.currentLayer?.title;
+          if (currentLayerTitle) {
+            const layerToSelect = layers.find(
+              (layer) => layer.title === currentLayerTitle,
+            );
+            if (layerToSelect) {
+              this.selectLayer(layerToSelect);
+            }
+          }
+        }),
+      ),
+    );
+
+    // Use the same stream to update attributesExcludedFromList
+    mainPanelStream.subscribe(() => {
+      this.hsFiltersService.attributesExcludedFromList = [
+        'hs_normalized_IDW_value',
+        'boundedBy',
+      ];
+    });
+  }
+
+  /**
+   * Selects a layer and updates the filter service
+   * @param layer The layer to select
+   */
+  async selectLayer(layer: HsLayerDescriptor | null) {
+    this.selectedLayer.set(layer);
+    if (!layer) {
+      return;
+    }
+
+    const wfsUrl = getWfsUrl(layer.layer);
```

Comment:
To ensure the WFS URL will be valid, I would trim any part after `?` mark, if it already exists in the `wfsUrl` property. For the sake of the following line.

Reply:
Good idea,  sure.  Will do.

Reply:
[6e16401](https://github.com/hslayers/hslayers-ng/pull/5223/commits/6e16401c6c4514c1cdb42c266479501a321b1dbb)